packages/netlify-cms-backend-github/src/API.ts

```diff
@@ -852,16 +852,16 @@ export default class API {
       const metadata = await this.retrieveMetadata(contentKey);
       // mark media files to remove
       const metadataMediaFiles: MediaFile[] = get(metadata, 'objects.files', []);
-      const mediaFilesToRemove: File[] = differenceBy(
+      const mediaFilesToRemove: { path: string; sha: string | null }[] = differenceBy(
         metadataMediaFiles,
         mediaFilesList,
         'path',
-      ).map(file => ({ ...file, remove: true, type: 'blob' }));
-      const branchData = await this.getBranch(branchName);
-      const changeTree = await this.updateTree(
```

Comment:
Instead of creating a new tree right away from the editorial workflow branch and then rebasing, I changed it so we first do the rebase then create a new tree on the rebased branch